app/src/main/java/org/thoughtcrime/securesms/contactshare/ContactModelMapper.java

```diff
@@ -0,0 +1,172 @@
+  }
+
+  public static Contact remoteToLocal(@NonNull SharedContact sharedContact) {
+    Name name = new Name(sharedContact.getName().getDisplay().orNull(),
+        sharedContact.getName().getGiven().orNull(),
+        sharedContact.getName().getFamily().orNull(),
+        sharedContact.getName().getPrefix().orNull(),
+        sharedContact.getName().getSuffix().orNull(),
+        sharedContact.getName().getMiddle().orNull());
+
+    List<Phone> phoneNumbers = new LinkedList<>();
+    if (sharedContact.getPhone().isPresent()) {
+      for (SharedContact.Phone phone : sharedContact.getPhone().get()) {
+        phoneNumbers.add(new Phone(phone.getValue(),
+                                   remoteToLocalType(phone.getType()),
+                                   phone.getLabel().orNull()));
+      }
+    }
+
+    List<Email> emails = new LinkedList<>();
+    if (sharedContact.getEmail().isPresent()) {
+      for (SharedContact.Email email : sharedContact.getEmail().get()) {
+        emails.add(new Email(email.getValue(),
+                             remoteToLocalType(email.getType()),
+                             email.getLabel().orNull()));
+      }
+    }
+
+    List<PostalAddress> postalAddresses = new LinkedList<>();
+    if (sharedContact.getAddress().isPresent()) {
+      for (SharedContact.PostalAddress postalAddress : sharedContact.getAddress().get()) {
+        postalAddresses.add(new PostalAddress(remoteToLocalType(postalAddress.getType()),
+                                              postalAddress.getLabel().orNull(),
+                                              postalAddress.getStreet().orNull(),
+                                              postalAddress.getPobox().orNull(),
+                                              postalAddress.getNeighborhood().orNull(),
+                                              postalAddress.getCity().orNull(),
+                                              postalAddress.getRegion().orNull(),
+                                              postalAddress.getPostcode().orNull(),
+                                              postalAddress.getCountry().orNull()));
+      }
+    }
+
+    Avatar avatar = null;
+    if (sharedContact.getAvatar().isPresent()) {
+      Attachment attachment = PointerAttachment.forPointer(Optional.of(sharedContact.getAvatar().get().getAttachment().asPointer())).get();
+      boolean    isProfile  = sharedContact.getAvatar().get().isProfile();
+
+      avatar = new Avatar(null, attachment, isProfile);
+    }
+
+    return new Contact(name, sharedContact.getOrganization().orNull(), phoneNumbers, emails, postalAddresses, avatar);
+  }
+
+  private static Phone.Type remoteToLocalType(SharedContact.Phone.Type type) {
+    switch (type) {
+      case HOME:   return Phone.Type.HOME;
+      case MOBILE: return Phone.Type.MOBILE;
+      case WORK:   return Phone.Type.WORK;
+      default:     return Phone.Type.CUSTOM;
+    }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 122:127, 131:136.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.